### DISCUSSION TITLE
[Bug] autocompletion does not work inside strings

### DISCUSSION THREAD

**Discussion by fdrag** [microsoft/monaco-editor#3679] [Q&A]
### Reproducible in vscode.dev or in VS Code Desktop?

- [X] Not reproducible in [vscode.dev](https://vscode.dev) or VS Code Desktop

### Reproducible in the monaco editor playground?

- [ ] Not reproducible in [the monaco editor playground](https://microsoft.github.io/monaco-editor/playground.html)

### Monaco Editor Playground Code

```typescript
monaco.editor.create(document.getElementById('container'), {
	value: "function hello() {\n\talert('Hello world!');\n}",
	language: 'typescript'
});
```

### Actual Behavior

![unknown](https://user-images.githubusercontent.com/97387068/148679731-e7bf786d-f0ac-4c0d-926d-70d750c15db5.png)

help

### Expected Behavior

show autocompletion

### Additional Context

if not bug then how to implement string completion? (sorry bad english )

**Comment by rcjsuen** [comment]
![image](https://user-images.githubusercontent.com/15629116/148680190-90fdfd38-992e-4c43-b4b9-8b5483a1ada9.png)

@fdrag Seems fine to me. 🤔

**Comment by fdrag** [comment]
> ![image](https://user-images.githubusercontent.com/15629116/148680190-90fdfd38-992e-4c43-b4b9-8b5483a1ada9.png)
> 
> @fdrag Seems fine to me. 🤔

https://user-images.githubusercontent.com/97387068/148680362-26ec3bc8-867f-4b0c-a9d2-711994c0f55d.mp4

**Comment by fdrag** [comment]
it still not working for me, how to implement string autocompletion myself?

i want to copy this 
![image](https://user-images.githubusercontent.com/97387068/148680482-d14ed935-ccaf-4686-85f8-b4c67f5f204e.png)

**Comment by rcjsuen** [comment]
> it still not working for me, how to implement string autocompletion myself?

If you invoke Intellisense explicitly with <kbd>Control</kbd>+<kbd>Space</kbd> (or <kbd>Cmd</kbd>+<kbd>Space</kbd> on macOS) it works.

> i want to copy this ![image](https://user-images.githubusercontent.com/97387068/148680482-d14ed935-ccaf-4686-85f8-b4c67f5f204e.png)

What is this from?

**Comment by fdrag** [comment]
> 

it is from visual studio code, im trying to making my own monaco editor for testing purposes but im not knowing how to implement those

how to automatic invoke intellisense when in qoutes?

**Comment by rcjsuen** [comment]
https://github.com/microsoft/monaco-editor/blob/a385674a34a33947483e0f7266b5a7dbcefe9b9d/src/typescript/languageFeatures.ts#L435-L438

1. At a surface level, the obvious issue is that the [trigger characters in Visual Studio Code](https://github.com/microsoft/vscode/blob/e7b3724e0cdc9095d73a06b47203baa2c1777ea8/extensions/typescript-language-features/src/languageFeatures/completions.ts#L642) are not the same as Monaco's (per above). We can see Monaco does not invoke Intellisense as aggressively as VS Code.

https://github.com/microsoft/monaco-editor/blob/a385674a34a33947483e0f7266b5a7dbcefe9b9d/website/typedoc/monaco.d.ts#L3056-L3060

2. Another issue is that `quickSuggestions` seems to be `false` when it supposedly defaults to being true. Turning it on explicitly seems to fix your problem. Please try the code below, @fdrag.

```TypeScript
const content = `interface env {
	a: "abc" | "def";
}
let env: env = {
	a: ""
}`;
monaco.editor.create(document.getElementById('container'), {
    value: content,
    language: 'typescript',
    quickSuggestions: true
});
```

**Reply by QingXia-Ela** [reply]
@rcjsuen Hello! I want to know it's there has any reason that need to let monaco suggestion completion only trigger by character `.`? I think it can trigger by more characters like `"` `'` (support string const completion) or `<` (support jsx component)

**Comment by rcjsuen** [comment]
```TypeScript
const defaults: ValidQuickSuggestionsOptions = {
    other: true,
    comments: false,
    strings: false
};
```
I think [this](https://github.com/microsoft/vscode/blob/b71b8a48ded639b56ca4027e41e863b904053659/src/vs/editor/common/config/editorOptions.ts#L1900-L1904) is the issue. Quick suggestions are enabled by default...they're just not enabled by default if the text caret is inside a comment or a string. The comment should be updated to reflect this in my opinion.

**Comment by fdrag** [comment]
> ```ts
> ValidQuickSuggestionsOptions 
> ```

Thank you for your explanation. :D
